keras/applications/densenet.py

```diff
@@ -0,0 +1,341 @@
+# -*- coding: utf-8 -*-
+"""DenseNet models for Keras.
+
+# Reference paper:
+
+- [Densely Connected Convolutional Networks](https://arxiv.org/abs/1608.06993) (CVPR 2017 Best Paper Award)
+
+# Reference implementation:
+
+- [Torch DenseNets](https://github.com/liuzhuang13/DenseNet/blob/master/models/densenet.lua)
+- [TensorNets](https://github.com/taehoonlee/tensornets/blob/master/tensornets/densenets.py)
+"""
+from __future__ import print_function
+from __future__ import absolute_import
+
+import os
+
+from .. import backend as K
+from ..models import Model
+from ..layers import Activation
+from ..layers import AveragePooling2D
+from ..layers import BatchNormalization
+from ..layers import Concatenate
+from ..layers import Conv2D
+from ..layers import Dense
+from ..layers import Flatten
+from ..layers import GlobalAveragePooling2D
+from ..layers import Input
+from ..layers import MaxPooling2D
+from ..layers import ZeroPadding2D
+from ..utils.data_utils import get_file
+from ..engine.topology import get_source_inputs
+from .imagenet_utils import decode_predictions
+from .imagenet_utils import _obtain_input_shape
+
+
+DENSENET121_WEIGHT_PATH = 'https://github.com/taehoonlee/deep-learning-models/releases/download/densenet/densenet121_weights_tf_dim_ordering_tf_kernels.h5'
+DENSENET121_WEIGHT_PATH_NO_TOP = 'https://github.com/taehoonlee/deep-learning-models/releases/download/densenet/densenet121_weights_tf_dim_ordering_tf_kernels_notop.h5'
+DENSENET169_WEIGHT_PATH = 'https://github.com/taehoonlee/deep-learning-models/releases/download/densenet/densenet169_weights_tf_dim_ordering_tf_kernels.h5'
+DENSENET169_WEIGHT_PATH_NO_TOP = 'https://github.com/taehoonlee/deep-learning-models/releases/download/densenet/densenet169_weights_tf_dim_ordering_tf_kernels_notop.h5'
+DENSENET201_WEIGHT_PATH = 'https://github.com/taehoonlee/deep-learning-models/releases/download/densenet/densenet201_weights_tf_dim_ordering_tf_kernels.h5'
+DENSENET201_WEIGHT_PATH_NO_TOP = 'https://github.com/taehoonlee/deep-learning-models/releases/download/densenet/densenet201_weights_tf_dim_ordering_tf_kernels_notop.h5'
+
+
+def dense(x, blocks, scope):
+    """A dense block.
+
+    # Arguments
+        x: input tensor.
+        blocks: integer, the number of building blocks.
+        scope: string, block label.
```

Comment:
All mentions of "scope" in this file should be renamed to `name`. Scope is TF terminology and does not make sense in this context (there are no variable scopes in Keras).